packages/client-cli/test/runtime.test.mjs

```diff
@@ -0,0 +1,104 @@
+import { request, moveToTmpdir } from './helper.js'
+import { test } from 'tap'
+import { join, dirname } from 'path'
+import { fileURLToPath } from 'node:url'
+import * as desm from 'desm'
+import { execa } from 'execa'
+import { cp, writeFile, readFile } from 'node:fs/promises'
+import split from 'split2'
+
+test('openapi client generation (javascript) via the runtime', async ({ teardown, comment, same, match }) => {
+  const dir = await moveToTmpdir(teardown)
+  comment(`working in ${dir}`)
+
+  await cp(join(dirname(fileURLToPath(import.meta.url)), 'fixtures', 'runtime'), dir, { recursive: true })
+
+  await writeFile(join(dir, 'services', 'somber-chariot', '.env'), `
+PLT_SERVER_HOSTNAME=127.0.0.1
+PORT=3003
+PLT_SERVER_LOGGER_LEVEL=info
+DATABASE_URL=sqlite://./db.sqlite
+`)
+
+  await writeFile(join(dir, 'services', 'languid-nobleman', '.env'), `
+PLT_SERVER_HOSTNAME=127.0.0.1
+PORT=3004
+PLT_SERVER_LOGGER_LEVEL=info
+`)
+
+  process.chdir(join(dir, 'services', 'languid-nobleman'))
+
+  await execa('node', [desm.join(import.meta.url, '..', 'cli.mjs'), '--name', 'movies', '--runtime', 'somber-chariot'])
+
+  const read = JSON.parse(await readFile(join(dir, 'services', 'languid-nobleman', 'platformatic.service.json'), 'utf8'))
+  match(read, {
+    clients: [{
+      serviceId: 'somber-chariot',
+      path: 'movies'
+    }]
+  })
+
+  const toWrite = `
+'use strict'
+
+module.exports = async function (app, opts) {
+  app.post('/', async (request, reply) => {
+    const res = await app.movies.createMovie({ title: 'foo' })
+    return res
+  })
+}
+`
+  await writeFile(join(dir, 'services', 'languid-nobleman', 'routes', 'movies.js'), toWrite)
+
+  process.chdir(dir)
+
+  const app2 = execa('node', [desm.join(import.meta.url, '..', '..', 'cli', 'cli.js'), 'start'])
+  teardown(() => app2.kill())
+
+  const stream = app2.stdout.pipe(split(JSON.parse))
+
+  // this is unfortuate :(
```

Comment:
If I'm doing my math right, this will be the 20th occurrence of this typo in the codebase 🤣. We should really create a set of test utils.